metagenomics.py

```diff
@@ -834,13 +868,18 @@ def diamond(inBam, db, taxDb, outReport, outReads=None, threads=None):
 
         if outReads is not None:
             # Interstitial save of stdout to output file
-            cmd += ' | tee >(gzip --best > {out})'.format(out=outReads)
+            reads_cmd = 'cat {inf} | tee >(gzip --best > {out}) > {pipe}'.format(
```

Comment:
we now have `pigz` as part of `requirements-conda.txt`, can you replace `gzip --best` with `pigz --best -p {threads}`?

Reply:
Also, I've never really experienced problems in the past with standard pipefitting of `Popen` objects (and I've never needed to use the `codecs` package for any of it)... I really think the shell-based pipes could be cleaned up at some point (doesn't have to be this PR though.. more for the future).

Reply:
My thoughts are that diamond is slow enough that even single threaded `gzip` can keep up with the output. The `codecs` is necessary for operating on pipes in py3 but it seems pipefitting can work if careful.